test/helpers/cilium.go

```diff
@@ -333,6 +350,93 @@ func (c *Cilium) ReportFailed(commands ...string) {
 		fmt.Fprint(wr, res.Output())
 	}
 	fmt.Fprint(wr, "StackTrace Ends\n")
+	c.ReportDump()
+}
+
+//ReportDump runs a variety of commands and writes the results to
+//testResultsPath
+func (c *Cilium) ReportDump() {
+	reportCmds := map[string]string{}
+
+	for cmd, log := range reportCommands {
+		reportCmds[fmt.Sprintf("sudo %s", cmd)] = log
+	}
+
+	reportEndpointsCommands := map[string]string{
+		"cilium endpoint get %s":         "endpoint_%s.txt",
+		"sudo cilium bpf policy list %s": "bpf_policy_list_%s.txt",
+	}
+
+	testPath, err := ReportDirectory()
+	if err != nil {
+		c.logCxt.WithError(err).Errorf(
+			"Cannot create Test results path '%s'", testPath)
+		return
+	}
+	reportMap(testPath, reportCmds, c.Node)
+
+	//Endpoint specific information:
+	eps, err := c.GetEndpointsIds()
+	if err != nil {
+		c.logCxt.Errorf("Cannot get endpoint ids")
+	}
+	for _, ep := range eps {
+		for cmd, logfile := range reportEndpointsCommands {
+			command := fmt.Sprintf(cmd, ep)
+			res := c.Node.Exec(command)
+
+			err = ioutil.WriteFile(
+				fmt.Sprintf("%s/%s", testPath, fmt.Sprintf(logfile, ep)),
+				res.CombineOutput().Bytes(),
+				0777)
+
+			if err != nil {
+				c.logCxt.WithError(err).Errorf("Cannot create test results for '%s'", command)
+			}
+		}
+	}
+
+	eps, err = c.GetEndpointsIdentityIds()
+	if err != nil {
+		c.logCxt.Errorf("Cannot get endpoint identity ids")
+	}
+	for _, ep := range eps {
+		res := c.Node.Exec(fmt.Sprintf("cilium identity get %s", ep))
+		err = ioutil.WriteFile(
+			fmt.Sprintf("%s/%s", testPath, fmt.Sprintf("identity_%s.txt", ep)),
+			res.CombineOutput().Bytes(),
+			0777)
+		if err != nil {
+			c.logCxt.WithError(err).Errorf("Cannot create test results for identity get")
+		}
+	}
+}
+
+//GatherLogs dumps Cilium, Cilium Docker, key-value store logs, and gops output
+//to the directory testResultsPath
+func (c *Cilium) GatherLogs() {
+	reportCmds := map[string]string{
+		"sudo journalctl -xe -u " + daemonName + " --no-pager":       "cilium.log",
+		"sudo journalctl -xe -u " + dockerDaemonName + " --no-pager": "cilium-docker.log",
+		"sudo docker logs cilium-consul":                             "consul.log",
+		`sudo bash -c "gops memstats $(pgrep ` + agentDaemon + `)"`:  "gops_memstats.txt",
+		`sudo bash -c "gops stack $(pgrep ` + agentDaemon + `)"`:     "gops_stack.txt",
+		`sudo bash -c "gops stats $(pgrep ` + agentDaemon + `)"`:     "gops_stats.txt",
+	}
+
+	gatherCommands := []string{
+		fmt.Sprintf("sudo cp -r %s /vagrant/%s/lib", runDir, testResultsPath),
+		fmt.Sprintf("sudo cp -r %s /vagrant/%s/run", libDir, testResultsPath),
+	}
+
+	reportMap(testResultsPath, reportCmds, c.Node)
+
+	for _, cmd := range gatherCommands {
+		res := c.Node.Exec(cmd)
+		if !res.WasSuccessful() {
+			c.logCxt.Errorf("cannot gather files for cmd '%s'", cmd)
```

Comment:
Change this log message to include why the command which was executed failed (maybe its stdout / stderr?) instead of just saying why it failed.